app/models/upload.rb

```diff
@@ -168,7 +168,7 @@ def self.secure_media_url?(url)
     # have secure-media-uploads in the URL e.g. /t/secure-media-uploads-are-cool/223452
     route = UrlHelper.rails_route_from_url(url)
     route[:action] == "show_secure" && route[:controller] == "uploads" && FileHelper.is_supported_media?(url)
-  rescue ActionController::RoutingError
+  rescue ActionController::RoutingError, Addressable::URI::InvalidURIError
```

Comment:
So it's `UrlHelper.rails_route_from_url` who's throwing this exception? Would it make more sense to have it handle it and return `nil` when the URL is invalid instead?